### ISSUE TITLE
false positive for catching-non-exception

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#125]
Originally reported by: **Anonymous**

---

when i create a class that inherits from socket.error (which inherits from Exception), and use the new class in a try/except block, i receive a pylint error message stating that i am "catching a exception which doesn't inherit from BaseException: <classname> (catching-non-exception)"

eg:

from socket import error

class MyError(error): pass

try:
  print 'a'
except MyError:
  print 'b'

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/125

**Comment by dhermes** [comment]
@PCManticore There may be a regression here, triggered in https://github.com/GoogleCloudPlatform/gcloud-python

To reproduce, install `grpcio==0.15.0`, then

```python
"""Bogus docstring."""


from grpc.framework.interfaces.face import face

try:
    MY_GLOBAL = 11
except face.NetworkError:
    raise ValueError('Other error')
```

then the error is

```
$ pylint foo.py
No config file found, using default configuration
************* Module foo
E:  8, 7: Catching an exception which doesn't inherit from BaseException: NetworkError (catching-non-exception)
```

----

But note that the MRO contains `BaseException` (and also `Exception` FWIW)

```python
>>> from grpc.framework.interfaces.face import face
D0718 16:12:55.983836848   26996 ev_posix.c:101]             Using polling engine: poll
>>> face.NetworkError.__mro__
(<class 'grpc.framework.interfaces.face.face.NetworkError'>, <class 'grpc.framework.interfaces.face.face.AbortionError'>, <type 'exceptions.Exception'>, <type 'exceptions.BaseException'>, <type 'object'>)
```

It's possible that there is some sort of failure in the static analysis caused by usage of `six.with_metaclass(abc.ABCMeta, Exception)` [in the parent][1]?

[1]: https://github.com/grpc/grpc/blob/f3f5c18e5ad14a35351f672181c51998b589cef3/src/python/grpcio/grpc/framework/interfaces/face/face.py#L101

**Comment by PCManticore** [comment]
@dhermes Should be fixed by https://github.com/PyCQA/pylint/commit/8d355577a0bc28c8ca6d94ec4723b7e9b818c047. Will release 1.6.4 later today or early tomorrow.

**Comment by dhermes** [comment]
@PCManticore You are awesome!
